logstash-core/src/test/java/org/logstash/ackedqueue/SettingsImplTest.java

```diff
@@ -0,0 +1,32 @@
+package org.logstash.ackedqueue;
+
+import junit.framework.TestCase;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SettingsImplTest {
+
+    @Test
+    public void checkpointRetrySetting() {
+        Settings settings =  SettingsImpl.fileSettingsBuilder("PATH/TO/Q")
+                .capacity(10)
+                .maxUnread(1024)
+                .queueMaxBytes(2147483647)
+                .checkpointMaxAcks(1)
+                .checkpointMaxWrites(1)
+                .checkpointRetry(true)
+                .elementClass(StringElement.class)
+                .build();
+
+        assertEquals(settings.getDirPath(), "PATH/TO/Q");
+        assertEquals(settings.getCapacity(), 10);
+        assertEquals(settings.getMaxUnread(), 1024);
+        assertEquals(settings.getQueueMaxBytes(), 2147483647);
+        assertEquals(settings.getCheckpointMaxAcks(), 1);
+        assertEquals(settings.getCheckpointMaxWrites(), 1);
+        assertTrue(settings.getCheckpointRetry());
```

Comment:
I think that a test named `checkpointRetrySetting` should check just that `checkpointRetry` assumes the expected value.
All the other asserts pollute the intention of this particular test.
Maybe you can split it in 2 tests:
- one to check all the settings
- one to check `checkpointRetry` specifically

or maybe, just change the test name in something more general, like: `verifySettingsGetConfiguredValues`

Reply:
I will change the name. Thanks for catching it.